gobblin-utility/src/test/java/org/apache/gobblin/util/WriterUtilsTest.java

```diff
@@ -93,13 +93,24 @@ public void testGetDataPublisherFinalOutputDir() {
 
   @Test
   public void testGetWriterFilePath() {
-    WorkUnit state = WorkUnit.createEmpty();
+    final String namespace = "dbNamespace";
+    final String table = "tableName";
+    Extract extract = new Extract(TableType.SNAPSHOT_ONLY, namespace, table);
+    WorkUnit state = WorkUnit.create(extract);
 
     state.setProp(ConfigurationKeys.WRITER_FILE_PATH, TEST_WRITER_FILE_PATH);
     Assert.assertEquals(WriterUtils.getWriterFilePath(state, 0, 0), TEST_WRITER_FILE_PATH);
 
     state.setProp(ConfigurationKeys.WRITER_FILE_PATH + ".0", TEST_WRITER_FILE_PATH);
     Assert.assertEquals(WriterUtils.getWriterFilePath(state, 1, 1), TEST_WRITER_FILE_PATH);
+
+    state.removeProp(ConfigurationKeys.WRITER_FILE_PATH);
+
+    state.setProp(ConfigurationKeys.WRITER_FILE_PATH_TYPE, "tablename");
+    Assert.assertEquals(WriterUtils.getWriterFilePath(state, 0, 0), new Path(table));
+
+    state.setProp(ConfigurationKeys.WRITER_FILE_PATH_TYPE, "db_table");
+    Assert.assertEquals(WriterUtils.getWriterFilePath(state, 0, 0), new Path(namespace + "/" + table));
```

Comment:
Use `Path.SEPARATOR` here since you are using `Path.SEPARATOR` when constructing the name.